src/nb_type.cpp

```diff
@@ -550,7 +594,7 @@ PyObject *nb_type_new(const type_data *t) noexcept {
     #undef COPY_FIELD
 
     if (temp_tp->tp_members) {
-        tp->tp_members = (PyMemberDef*)((char *)tp + tp->tp_basicsize);
+        tp->tp_members = (PyMemberDef*)((char *)tp + Py_TYPE(tp)->tp_basicsize);
```

Comment:
I realize my last PR had a tiny bug: we should be using the basicsize of the type itself.

Reply:
Fixed here: https://github.com/wjakob/nanobind/commit/6a25ef398b2fd91d82c2241ff42b482843d2b130